sparta.go

```diff
@@ -933,6 +943,20 @@ func NewLambda(roleNameOrIAMRoleDefinition interface{},
 	return lambda
 }
 
+// NewNamedLambda returns a LambdaAWSInfo value with a declared function name that can be provisioned
+// via CloudFormation. The `functionName` value must be a non-empty string, otherwise CloudFormation
+// will error out as "" is not a valid function name. The roleNameOrIAMRoleDefinition must either be
+// a `string` or `IAMRoleDefinition` type
+func NewNamedLambda(roleNameOrIAMRoleDefinition interface{},
+fn LambdaFunction,
+lambdaOptions *LambdaFunctionOptions,
+functionName string) *LambdaAWSInfo {
```

Comment:
That would be fine,but we still need to check in the export function for a
name before adding it the lambda resource.

Sean

On Tue, Aug 23, 2016 at 10:03 PM, Matt Weagle <notifications@github.com>
wrote:

> In sparta.go
> <https://github.com/mweagle/Sparta/pull/12#discussion_r75995194>:
>
> > @@ -933,6 +943,20 @@ func NewLambda(roleNameOrIAMRoleDefinition interface{},
> >  	return lambda
> >  }
> >
> > +// NewNamedLambda returns a LambdaAWSInfo value with a declared function name that can be provisioned
> > +// via CloudFormation. The `functionName` value must be a non-empty string, otherwise CloudFormation
> > +// will error out as "" is not a valid function name. The roleNameOrIAMRoleDefinition must either be
> > +// a `string` or `IAMRoleDefinition` type
> > +func NewNamedLambda(roleNameOrIAMRoleDefinition interface{},
> > +fn LambdaFunction,
> > +lambdaOptions *LambdaFunctionOptions,
> > +functionName string) *LambdaAWSInfo {
>
> Thoughts on changing this to (*LambdaAWSInfo, error) to catch empty
> functionName values prior to the CloudFormation marshalling step?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mweagle/Sparta/pull/12/files/821b06cc9b8fe79151f97cb776324451af16c7c2#r75995194>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AUKsljH-UOQSptcTdmPaqFHEfNdII-cMks5qi9CTgaJpZM4JrlgX>
> .
>